app/src/main/java/com/asterisk/nam/demobroadcastintentpermission/MainActivity.java

```diff
@@ -1,13 +1,91 @@
 package com.asterisk.nam.demobroadcastintentpermission;
 
-import android.support.v7.app.AppCompatActivity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.Uri;
+import android.net.wifi.WifiManager;
 import android.os.Bundle;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.Switch;
 
 public class MainActivity extends AppCompatActivity {
 
+
+    private final static Integer AGE_CAT = 3;
+    public final static String KEY_INTENT_NORMAL = "my_string";
+    private final static String DATA_INTENT_NORMAL = "Today, The Weather is very cold";
+    public final static String KEY_INTENT_BUNDLE = "my_string_bundle";
+    private final static String DATA_INTENT_BUNDLE = "Today, my laptop is repaired";
+    public final static String KEY_INTENT_BUNDLE_PACERABLE = "my_string_bundle_pacerable";
+    private Button mButtonSendIntentEx,mButtonSendIntentView,mButtonSendIntentMail;
+    private Cat mCat;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        init();
+        clickButton();
+    }
+
+    public void init() {
+        mButtonSendIntentEx = findViewById(R.id.button_send_intent_ex);
+        mButtonSendIntentView = findViewById(R.id.button_send_intent_view);
+        mButtonSendIntentMail = findViewById(R.id.button_send_intent_mail);
+    }
+
+    public void clickButton() {
```

Comment:
nên tạo 1 phương thức lắng nghe sự kiện, và các button hay textView nên implement setOnClickListener(this) thay vì tạo new